server/conn.go

```diff
@@ -1025,7 +1019,13 @@ func (cc *clientConn) dispatch(ctx context.Context, data []byte) error {
 	cc.lastPacket = data
 	cmd := data[0]
 	data = data[1:]
-	if variable.EnablePProfSQLCPU.Load() {
+	if config.TopSQLEnabled() {
+		normalizedSQL, digest := getLastStmtInConn{cc}.pprofLabelNormalizedAndDigest()
```

Comment:
`pprofLabelNormalizedAndDigest` parses the `data` in `cc`, instead of reuse some processed result. This makes parsing cost doubled when top sql is enabled?

Reply:
Yes. Maybe we can put this in another place to avoid the double cost of parsing.

Reply:
How about put it at some place where SQL has been parsed and digest is already generated?

Reply:
Already remove this, PTAL